### PR TITLE
bugfix: fix the exception with SeataTCCFenceAutoConfiguration

### LINKED ISSUE #4514: SeataTCCFenceAutoConfiguration throws error when database is not in use
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
当未使用数据库时SeataTCCFenceAutoConfiguration 会出现 ClaasNotFound 错误，因为这个类强依赖了PlatformTransactionManager。
SeataTCCFenceAutoConfiguration throws ClaasNotFoundException when database is not in use, Because it strongly depends on PlatformTransactionManager.

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. 客户端不依赖jdbc和connector
2. 使用seata1.5版本启动客户端

### Ⅵ. Environment:

- JDK version : 1.8
- Seata version:  1.5.0
- OS : 
- Others:

### PR DESCRIPTION
**Pull request by pengten**
fix the ClassNotFoundException with SeataTCCFenceAutoConfiguration when jdbc unused.

- [x] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
修复SeataTCCFenceAutoConfiguration在客户端未使用DB时，启动抛出ClassNotFoundException的问题。

### Ⅱ. Does this pull request fix one issue?

fixes #4514

### CODE REVIEW

File: seata-spring-autoconfigure/seata-spring-autoconfigure-client/src/main/java/io/seata/spring/boot/autoconfigure/SeataTCCFenceAutoConfiguration.java [hunk 1]
```diff
@@ -36,6 +37,7 @@
  * @author kaka2code
  */
 @ConditionalOnExpression("${seata.enabled:true}")
+@ConditionalOnClass(name = {"org.springframework.transaction.PlatformTransactionManager"})
```
Comment (a364176773) [inline seata-spring-autoconfigure/seata-spring-autoconfigure-client/src/main/java/io/seata/spring/boot/autoconfigure/SeataTCCFenceAutoConfiguration.java:40]: ConditionalOnBean 采用beanname形式，下方不要ConditionalOnBean了可以吗？

**Comment by clverhack** [comment]
这个问题再1.5.1 里面解决了吗

**Review by slievrly — APPROVED** [review]
LGTM,  Pls remove change log into 1.5.2.

**Review by a364176773 — APPROVED** [review]
LGTM 同上

**Pull request merged**